### PR TITLE
feat: Add Modbus scalability testing

### LINKED ISSUE #137: Modbus scalability test
Measure how many devices can connect to device service.

### PR DESCRIPTION
**Pull request by weichou1229**
- Add testing script and modify the docker-compose file to run the Modbus scalability testing.
- Write the document `docs/modbus-scalability-test.md`

Fix #137

### CODE REVIEW

**Review by cherrycl — APPROVED** [review]
LGTM

File: docs/modbus-scalability-test.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+- docker-compose
+
+## Prepare Test Environment
+
+### 1. Deploy southbound Modbus simulator
+Run the following command to deploy simulators:
+```
+docker run --rm -d --network host --name modbus-simulator \
+    -e SIMULATOR_NUMBER=1000 -e STARTING_PORT=10000 \
+    nexus3.edgexfoundry.org:10003/edgex-devops/edgex-modbus-simulator
```
Comment (jinlinGuan) [inline docs/modbus-scalability-test.md:20]: Could you please add an annotation for the arm64 image?

File: docs/modbus-scalability-test.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    MQTT_BROKER_IP=${replace-with-broker-ip}
+    ```
+    - MQTT_BROKER_IP: The northbound simulator host IP
+   
+### 4. Export the environment variables
+```
+export ARCH=x86_64
+export SECURITY_SERVICE_NEEDED=false
+export COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose
+export EDGEX_TAF_IMAGE=nexus3.edgexfoundry.org:10003/docker-edgex-taf-common
```
Comment (jinlinGuan) [inline docs/modbus-scalability-test.md:66]: Also need arm64 image annotation. Thanks.

File: docs/modbus-scalability-test.md (hunk 1 above)
Comment (weichou1229) [reply docs/modbus-scalability-test.md:21]: Added.

File: docs/modbus-scalability-test.md (hunk 2 above)
Comment (weichou1229) [reply docs/modbus-scalability-test.md:74]: Added.

File: docs/modbus-scalability-test.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+# arm64
+export ARCH=arm64
+export COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose-arm64:latest
+export EDGEX_TAF_IMAGE=nexus3.edgexfoundry.org:10003/docker-edgex-taf-common-arm64:latest
+```
+
+### 5. Fetch the latest docker-compose file
+```
+cd edgex-taf/TAF/utils/scripts/docker
+sh get-compose-file.sh -redis x86_64 -
```
Comment (jinlinGuan) [inline docs/modbus-scalability-test.md:85]:
```suggestion
sh get-compose-file.sh -redis ${ARCH} -
```

Comment (weichou1229) [reply docs/modbus-scalability-test.md]: Modified.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
this issue https://github.com/edgexfoundry/go-mod-messaging/issues/76 would impact this test

File: TAF/utils/scripts/docker/docker-compose-end-to-end.yaml [hunk 4]
```diff
@@ -117,3 +117,28 @@
       - data
     networks:
       - edgex-network
+
+  scalability-test-mqtt-export:
```
Comment (cherrycl) [inline TAF/utils/scripts/docker/docker-compose-end-to-end.yaml:121]: Needs to add the depends_on.
```
depends_on:
      - consul
      - data
```

File: TAF/utils/scripts/docker/start-services.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+CONF_DIR=/custom-config
+docker run --rm -v ${WORK_DIR}:${WORK_DIR} -w ${WORK_DIR} -v /var/run/docker.sock:/var/run/docker.sock \
+        --env WORK_DIR=${WORK_DIR} --env PROFILE=${PROFILE} --security-opt label:disable \
+        ${COMPOSE_IMAGE} -f "${WORK_DIR}/TAF/utils/scripts/docker/docker-compose.yaml" start $*
```
Comment (cherrycl) [inline TAF/utils/scripts/docker/start-services.sh:6]: When do the scripts use this file?

File: TAF/utils/scripts/docker/docker-compose-end-to-end.yaml (hunk 4 above)
Comment (weichou1229) [reply TAF/utils/scripts/docker/docker-compose-end-to-end.yaml:121]: Added.

File: TAF/utils/scripts/docker/start-services.sh (hunk 5 above)
Comment (weichou1229) [reply TAF/utils/scripts/docker/start-services.sh]: Removed the redundant script.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testCaseModules/keywords/setup/edgex.py [hunk 6]
```diff
@@ -32,15 +33,6 @@ def shutdown_services(*args):
     run_command(cmd)
 
 
-def stop_services(*args):
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply TAF/testCaseModules/keywords/setup/edgex.py]: Thanks for catching that. Added it back.

**Review by jinlinGuan — APPROVED** [review]
LGTM

File: TAF/utils/scripts/docker/docker-compose-end-to-end.yaml [hunk 7]
```diff
@@ -18,11 +18,11 @@
       EDGEX_SECURITY_SECRET_STORE: "false"
       Writable_Pipeline_Functions_HTTPPostJSON_Parameters_url: http://${DOCKER_HOST_IP}:7770
       Writable_LogLevel: DEBUG
+    networks:
+      - edgex-network
     depends_on:
       - consul
       - data
-    networks:
```
Comment (jinlinGuan) [inline TAF/utils/scripts/docker/docker-compose-end-to-end.yaml:24]: Should not change the order.

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/utils/scripts/docker/docker-compose-end-to-end.yaml (hunk 7 above)
Comment (weichou1229) [reply TAF/utils/scripts/docker/docker-compose-end-to-end.yaml]: Modified.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**